### PR TITLE
[SPARK-29018][SQL][phase3] Remove Hive deps for new thrift server

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
This PR removes Hive deps for new thrift server.

### How was this patch tested?
Unit test and manual test:
```scala
./dev/make-distribution.sh  --tgz   -Phive -Pspark-thriftserver -Phive-provided
```

### CODE REVIEW

File: sql/jdbc/src/test/scala/org/apache/spark/sql/jdbc/UISeleniumSuite.scala [hunk 1]
```diff
@@ -65,12 +65,12 @@ class UISeleniumSuite
...
-        |  --hiveconf ${ConfVars.METASTORECONNECTURLKEY}=$metastoreJdbcUri
-        |  --hiveconf ${ConfVars.METASTOREWAREHOUSE}=$warehousePath
-        |  --hiveconf ${ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST}=localhost
-        |  --hiveconf ${ConfVars.HIVE_SERVER2_TRANSPORT_MODE}=$mode
-        |  --hiveconf $portConf=$port
+        |  --sparkconf hive.root.logger=INFO,console
+        |  --sparkconf ${ConfVars.METASTORECONNECTURLKEY}=$metastoreJdbcUri
+        |  --sparkconf ${ConfVars.METASTOREWAREHOUSE}=$warehousePath
+        |  --sparkconf ${ConfVars.HIVE_SERVER2_THRIFT_BIND_HOST}=localhost
+        |  --sparkconf ${ConfVars.HIVE_SERVER2_TRANSPORT_MODE}=$mode
```
Comment (wangyum) [inline sql/jdbc/src/test/scala/org/apache/spark/sql/jdbc/UISeleniumSuite.scala]: `--sparkconf` -> `--conf`?

File: sql/service/src/main/scala/org/apache/spark/sql/service/internal/ServiceConf.scala [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+      .intConf
+      .createWithDefault(500)
+
+
+  val THRIFTSERVER_THRIFT_LOGIN_BEBACKOFF_SLOT_LENGTH =
+    buildConf("spark.sql.thriftserver.thrift.exponential.backoff.slot.length")
+      .internal()
+      .doc("Binary exponential backoff slot time for Thrift clients during login to SparkServer2," +
+        "for retries until hitting Thrift client timeout")
+      .timeConf(TimeUnit.MILLISECONDS)
```
Comment (wangyum) [inline sql/service/src/main/scala/org/apache/spark/sql/service/internal/ServiceConf.scala]: `.timeConf(TimeUnit.MILLISECONDS)` -> `intConf `?

File: sql/jdbc/src/test/scala/org/apache/spark/sql/jdbc/SparkThriftServer2Suites.scala [hunk 3]
```diff
@@ -1036,11 +1037,11 @@ abstract class SparkThriftServer2Test extends SparkFunSuite with BeforeAndAfterA
        |  --master local
        |  --hiveconf ${ConfVars.METASTORECONNECTURLKEY}=$metastoreJdbcUri
        |  --hiveconf ${ConfVars.METASTOREWAREHOUSE}=$warehousePath
```
Comment (wangyum) [inline sql/jdbc/src/test/scala/org/apache/spark/sql/jdbc/SparkThriftServer2Suites.scala]: `--hiveconf ${ConfVars.METASTOREWAREHOUSE}=$warehousePath` -> `--conf ${StaticSQLConf.WAREHOUSE_PATH.key}=$warehousePath`?



`--hiveconf ${ConfVars.METASTORECONNECTURLKEY}=$metastoreJdbcUri` -> `--driver-java-options--driver-java-options=-Dderby.system.home=$metastoreJdbcUri`?

File: sql/service/src/main/java/org/apache/spark/sql/service/cli/thrift/ThriftCLIService.java [hunk 4]
```diff
@@ -128,49 +131,46 @@ public void processContext(ServerContext serverContext,
...
-      workerKeepAliveTime =
-          hiveConf.getTimeVar(ConfVars.HIVE_SERVER2_THRIFT_WORKER_KEEPALIVE_TIME, TimeUnit.SECONDS);
-      portString = System.getenv("HIVE_SERVER2_THRIFT_PORT");
+      workerKeepAliveTime =(long) sqlConf.getConf(ServiceConf.THRIFTSERVER_THRIFT_WORKER_KEEPALIVE_TIME());
+      portString = System.getenv("SPARK_THRIFTSERVER_THRIFT_PORT");
       if (portString != null) {
         portNum = Integer.valueOf(portString);
       } else {
-        portNum = hiveConf.getIntVar(ConfVars.HIVE_SERVER2_THRIFT_PORT);
+        portNum = (int) sqlConf.getConf(ServiceConf.THRIFTSERVER_HTTP_PORT());
```
Comment (wangyum) [inline sql/service/src/main/java/org/apache/spark/sql/service/cli/thrift/ThriftCLIService.java]: `THRIFTSERVER_HTTP_PORT ` -> `THRIFTSERVER_THRIFT_PORT`?

File: sql/service/src/main/scala/org/apache/spark/sql/service/cli/session/SparkSQLSessionManager.scala [hunk 5]
```diff
@@ -67,9 +70,8 @@ private[service] class SparkSQLSessionManager(sparkServer: SparkServer2, sqlCont
       sqlContext.newSession()
     }
     ctx.setConf(HiveUtils.FAKE_HIVE_VERSION.key, HiveUtils.builtinHiveVersion)
-    val hiveSessionState = session.getSessionState
-    setConfMap(ctx, hiveSessionState.getOverriddenConfigurations)
-    setConfMap(ctx, hiveSessionState.getHiveVariables)
+    setConfMap(ctx, session.getVariables)
+    setConfMap(ctx, session.getSQLConf.getAllConfs.asJava)
```
Comment (wangyum) [inline sql/service/src/main/scala/org/apache/spark/sql/service/cli/session/SparkSQLSessionManager.scala]: Could we comment out these changes for now?
```scala
org.apache.spark.sql.AnalysisException: Cannot modify the value of a Spark config: spark.driver.host;
        at org.apache.spark.sql.RuntimeConfig.requireNonStaticConf(RuntimeConfig.scala:158)
        at org.apache.spark.sql.RuntimeConfig.set(RuntimeConfig.scala:40)
        at org.apache.spark.sql.SQLContext.setConf(SQLContext.scala:114)
        at org.apache.spark.sql.service.cli.session.SparkSQLSessionManager.setConfMap(SparkSQLSessionManager.scala:93)
        at org.apache.spark.sql.service.cli.session.SparkSQLSessionManager.openSession(SparkSQLSessionManager.scala:74)
        at org.apache.spark.sql.service.cli.CLIService.openSessionWithImpersonation(CLIService.java:166)
        at org.apache.spark.sql.service.cli.thrift.ThriftCLIService.getSessionHandle(ThriftCLIService.java:354)
        at org.apache.spark.sql.service.cli.thrift.ThriftCLIService.OpenSession(ThriftCLIService.java:248)
        at org.apache.spark.sql.service.rpc.thrift.TCLIService$Processor$OpenSession.getResult(TCLIService.java:1376)
        at org.apache.spark.sql.service.rpc.thrift.TCLIService$Processor$OpenSession.getResult(TCLIService.java:1361)

```

File: sql/service/src/main/java/org/apache/spark/sql/service/cli/thrift/ThriftCLIService.java (hunk 4 above)
Comment (AngersZhuuuu) [reply sql/service/src/main/java/org/apache/spark/sql/service/cli/thrift/ThriftCLIService.java]: > `THRIFTSERVER_HTTP_PORT ` -> `THRIFTSERVER_THRIFT_PORT`?

Updated

File: sql/service/src/main/scala/org/apache/spark/sql/service/internal/ServiceConf.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply sql/service/src/main/scala/org/apache/spark/sql/service/internal/ServiceConf.scala]: > `.timeConf(TimeUnit.MILLISECONDS)` -> `intConf `?

In hive is Time Type

File: sql/jdbc/src/test/scala/org/apache/spark/sql/jdbc/SparkThriftServer2Suites.scala (hunk 3 above)
Comment (AngersZhuuuu) [reply sql/jdbc/src/test/scala/org/apache/spark/sql/jdbc/SparkThriftServer2Suites.scala]: Done

**Comment by wangyum** [comment]
Could we update `sql/service/pom.xml` to ?
```diff
diff --git a/sql/service/pom.xml b/sql/service/pom.xml
index ec2d137961..90e930b12b 100644
--- a/sql/service/pom.xml
+++ b/sql/service/pom.xml
@@ -37,7 +37,7 @@
     <dependencies>
         <dependency>
             <groupId>org.apache.spark</groupId>
-            <artifactId>spark-hive_${scala.binary.version}</artifactId>
+            <artifactId>spark-sql_${scala.binary.version}</artifactId>
             <version>${project.version}</version>
         </dependency>
         <dependency>
@@ -55,104 +55,13 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.apache.spark</groupId>
-            <artifactId>spark-hive_${scala.binary.version}</artifactId>
-            <version>${project.version}</version>
-            <type>test-jar</type>
-            <scope>test</scope>
+            <groupId>org.apache.thrift</groupId>
+            <artifactId>libthrift</artifactId>
         </dependency>
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
         </dependency>
-        <dependency>
-            <groupId>${hive.group}</groupId>
-            <artifactId>hive-cli</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>${hive.group}</groupId>
-            <artifactId>hive-beeline</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>${hive.group}</groupId>
-            <artifactId>hive-service</artifactId>
-            <version>${hive.version}</version>
-            <scope>${hive.deps.scope}</scope>
-            <exclusions>
-                <exclusion>
-                    <groupId>${hive.group}</groupId>
-                    <artifactId>hive-exec</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>${hive.group}</groupId>
-                    <artifactId>hive-metastore</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.eclipse.jetty.aggregate</groupId>
-                    <artifactId>jetty-all</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>commons-codec</groupId>
-                    <artifactId>commons-codec</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>commons-cli</groupId>
-                    <artifactId>commons-cli</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>commons-lang</groupId>
-                    <artifactId>commons-lang</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>commons-logging</groupId>
-                    <artifactId>commons-logging</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.apache.thrift</groupId>
-                    <artifactId>libfb303</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.apache.thrift</groupId>
-                    <artifactId>libthrift</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.apache.curator</groupId>
-                    <artifactId>curator-framework</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.apache.curator</groupId>
-                    <artifactId>curator-recipes</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>jdk.tools</groupId>
-                    <artifactId>jdk.tools</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>${hive.group}</groupId>
-                    <artifactId>hive-service-rpc</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>${hive.group}</groupId>
-                    <artifactId>hive-llap-server</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.eclipse.jetty.aggregate</groupId>
-                    <artifactId>jetty-all</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>tomcat</groupId>
-                    <artifactId>jasper-compiler</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>tomcat</groupId>
-                    <artifactId>jasper-runtime</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>org.jamon</groupId>
-                    <artifactId>jamon-runtime</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
         <dependency>
             <groupId>org.eclipse.jetty</groupId>
             <artifactId>jetty-server</artifactId>

```

File: sql/service/src/main/java/org/apache/spark/sql/service/auth/shims/HadoopShimsSecure.java [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+  private static String[] dedup(String[] locations) throws IOException {
+    Set<String> dedup = new HashSet<String>();
+    Collections.addAll(dedup, locations);
+    return dedup.toArray(new String[dedup.size()]);
+  }
+
+  @Override
+  public void checkFileAccess(FileSystem fs, FileStatus stat, FsAction action)
+      throws IOException, AccessControlException, Exception {
+    DefaultFileAccess.checkFileAccess(fs, stat, action);
```
Comment (wangyum) [inline sql/service/src/main/java/org/apache/spark/sql/service/auth/shims/HadoopShimsSecure.java]: `DefaultFileAccess` is a Hive class.

**Pull request merged**